### DISCUSSION TITLE
lsp completion for html and css not working.

### DISCUSSION THREAD

**Discussion by saman-zdf** [LazyVim/LazyVim#259] [Q&A] [answered]
Hi, I've started using lazyvim and so fay I really enjoy the experience, I also add a few other plugin I used to had with my neovim setup and it works fine. The only problem I've been facing and cannot resolve is I don't get the completion html tag and css property list when I type the tags or css props. Is there any way anyone could help me out to configure the lsp set up  to make those completion work? Thanks in advance  
This is my repo: 
https://github.com/saman-zdf/lazyvim_configuation

**Comment by folke** [comment]
Did you install an lsp server to handle css / html? Otherwise that won't just work. Only json and luals are installed by default.

**Reply by saman-zdf** [reply]
Thanks for your respond, would it be possible to let me know how can I install the lsp server, I would really appreciate your help. Thanks

**Comment by dpetka2001** [comment, marked as answer]
Just use `mason` to install the appropriate server that you want. You might also want to take a look at [`LSP docs`](https://www.lazyvim.org/plugins/lsp#nvim-lspconfig).

**Reply by saman-zdf** [reply]
Thanks.

**Reply by dpetka2001** [reply]
```lua
return { 
   -- Add `pyright` to mason 
   { 
     "williamboman/mason.nvim", 
     opts = function(_, opts) 
       vim.list_extend(opts.ensure_installed, { "pyright", "black" }) 
     end, 
   }, 
  
   -- Setup null-ls with `black` 
   { 
     "jose-elias-alvarez/null-ls.nvim", 
     opts = function(_, opts) 
       local nls = require("null-ls") 
       opts.sources = vim.list_extend(opts.sources, { nls.builtins.formatting.black }) 
     end, 
   }, 
 }
```
This is my config for python. I use `pyright` as LSP server and `black` as formatter. Just change these values to what you want to use. Unfortunately i don't use html or css and don't know what to suggest you to use. If you don't know which LSP server to use maybe you should make a new post under general discussion to ask other people for suggestions. Or even neovim subreddit.

**Reply by saman-zdf** [reply]
@dpetka2001 Thanks for your help mate, I was able to fix it. Much appreciate it.
